catalogs/catalog-fileset-hadoop/src/main/java/com/datastrato/gravitino/catalog/file/HadoopCatalogOperations.java

```diff
@@ -0,0 +1,383 @@
+        .withAuditInfo(schemaEntity.auditInfo())
+        .build();
+  }
+
+  @Override
+  public Schema loadSchema(NameIdentifier ident) throws NoSuchSchemaException {
+    try {
+      SchemaEntity schemaEntity = store.get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
+
+      return new HadoopSchema.Builder()
+          .withName(ident.name())
+          .withComment(schemaEntity.comment())
+          .withProperties(schemaEntity.properties())
+          .withAuditInfo(schemaEntity.auditInfo())
+          .build();
+
+    } catch (NoSuchEntityException exception) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", exception);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load schema " + ident.name(), ioe);
+    }
+  }
+
+  @Override
+  public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
+      throws NoSuchSchemaException {
+    try {
+      if (!store.exists(ident, Entity.EntityType.SCHEMA)) {
+        throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist");
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to check if schema " + ident.name() + " exists", ioe);
+    }
+
+    try {
+      SchemaEntity entity =
+          store.update(
+              ident,
+              SchemaEntity.class,
+              Entity.EntityType.SCHEMA,
+              schemaEntity -> updateSchemaEntity(ident, schemaEntity, changes));
+
+      return new HadoopSchema.Builder()
+          .withName(ident.name())
+          .withComment(entity.comment())
+          .withProperties(entity.properties())
+          .withAuditInfo(entity.auditInfo())
+          .build();
+
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to update schema " + ident.name(), ioe);
+    } catch (NoSuchEntityException nsee) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", nsee);
+    } catch (AlreadyExistsException aee) {
+      throw new RuntimeException("Schema " + ident.name() + " already exists", aee);
+    }
+  }
+
+  @Override
+  public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
+    try {
+      SchemaEntity schemaEntity = store.get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
+      Map<String, String> properties =
+          Optional.ofNullable(schemaEntity.properties()).orElse(Collections.emptyMap());
+
+      String schemaLocation =
+          (String)
+              SCHEMA_PROPERTIES_METADATA.getOrDefault(
+                  properties, HadoopSchemaPropertiesMetadata.LOCATION);
+      Path schemaPath =
+          Optional.ofNullable(schemaLocation)
+              .map(Path::new)
+              .orElse(catalogStorageLocation.map(p -> new Path(p, ident.name())).orElse(null));
+
+      // Nothing to delete if the schema path is not set.
+      if (schemaPath == null) {
+        return true;
```

Comment:
The return value is not consistent with the interface description `@return True if the schema exists and is dropped successfully, false otherwise.`

Reply:
`schemaPath` is an optional property, we can have a schema object without `schemaPath`, so this can be null and we should delete the metadata. The semantic is correct.